include/exadg/utilities/enum_patterns.h

```diff
@@ -0,0 +1,82 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2022 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_EXADG_UTILITIES_ENUM_PATTERNS_H_
+#define INCLUDE_EXADG_UTILITIES_ENUM_PATTERNS_H_
+
+// inspired by following code which uses different backend for enums
+// https://github.com/peterrum/dealii-parameter-handler-enum/blob/master/include/deal.II/base/patterns_enum.h
```

Comment:
I extended the documentation in [c3f0a7d](https://github.com/exadg/exadg/pull/382/commits/c3f0a7daf7857d39dcd9474eb86602d5bde96c1f) as suggested. I still want to cite the project since it helped a lot during development of the code.